src/Sentry/PlatformAbstractions/FrameworkInfo.NetFx.cs

```diff
@@ -80,31 +80,22 @@ public static partial class FrameworkInfo
             return latest;
         }
 
-        internal static RegistryKey? TryOpenSubKey(string name, out Exception? exception)
-        {
-            try
-            {
-                var subKey = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry32)
-                    .OpenSubKey(name);
-                exception = null;
-                return subKey;
-            }
-            catch (Exception ex)
-            {
-                exception = ex;
-                return null;
-            }
-        }
-
         /// <summary>
         /// Get all .NET Framework installations in this machine
         /// </summary>
         /// <seealso href="https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed#to-find-net-framework-versions-by-querying-the-registry-in-code-net-framework-1-4"/>
         /// <returns>Enumeration of installations</returns>
         public static IEnumerable<FrameworkInstallation> GetInstallations()
         {
-            using var ndpKey = TryOpenSubKey(NetFxNdpRegistryKey, out _);
-            return GetInstallationsFromRegistryKey(ndpKey);
+            try
+            {
+                using var ndpKey = Registry.LocalMachine.OpenSubKey(NetFxNdpRegistryKey, false);
+                return GetInstallationsFromRegistryKey(ndpKey);
```

Comment:
`ndpKey` is disposed by the time the enumerator actually enumerates. The issue is that this statements returns immediately, but the `foreach` inside `GetInstallationsFromRegistryKey()` kicks in after `ndpKey` is disposed.

You might need to do this:

```suggestion
                using var ndpKey = Registry.LocalMachine.OpenSubKey(NetFxNdpRegistryKey, false);
                foreach (var installation in GetInstallationsFromRegistryKey(ndpKey))
                {
                    yield return installation;
                }
```

Reply:
Or, alternatively:

```suggestion
                using var ndpKey = Registry.LocalMachine.OpenSubKey(NetFxNdpRegistryKey, false);
                return GetInstallationsFromRegistryKey(ndpKey).ToArray();
```

Reply:
It's weird that it worked on my Machine(TM), but at least, the test pointed out it was broken 😅